src/neptune/metadata_containers/model_version.py

```diff
@@ -222,8 +222,11 @@ def _get_or_create_api_object(self) -> ApiExperiment:
             if self._mode == Mode.READ_ONLY:
                 raise NeedExistingModelVersionForReadOnlyMode()
 
+            model_id = self._project_api_object.sys_id
+            full_model_name = self._model if self._model.split("-")[0] == model_id else f"{model_id}-{self._model}"
```

Comment:
> `self._model.split("-")[0] == model_id`

How about just using `startswith(...)`?

Reply:
What if someone has a model named `ABC-ABCModel`?
Then the bare model name starts with `ABC`, but doesn't contain the ID that backend expects